code/modules/mob/living/carbon/human/life.dm

```diff
@@ -1627,7 +1635,8 @@
 	if (!exhaust_threshold) // Also quit if there's no exhaust threshold specified, because division by 0 is amazing.
 		return
 
-	if (failed_last_breath || oxyloss > exhaust_threshold)//Can't catch our breath if we're suffocating
+	if (failed_last_breath || (oxyloss + halloss) > exhaust_threshold)//Can't catch our breath if we're suffocating

```

Comment:
This is going to be troublesome, perhaps. You should check what happens if you baton yourself into paincrit, can you breathe then?

Reply:
This is only for stamina replenishment.